pkg/envoy/cds/response.go

```diff
@@ -34,6 +23,21 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 		return nil, err
 	}
 
+	if proxy.Kind() == envoy.KindGateway {
+		// Build remote clusters based on allowed outbound services
+		for _, dstService := range meshCatalog.ListAllowedOutboundServicesForIdentity(proxyIdentity.ToServiceIdentity()) {
+			cluster, err := getUpstreamServiceCluster(proxyIdentity.ToServiceIdentity(), dstService, cfg, NoPermissive, NoTLS)
+			if err != nil {
+				log.Error().Err(err).Msgf("Failed to construct service cluster for service %s for proxy with XDS Certificate SerialNumber=%s on Pod with UID=%s",
+					dstService.Name, proxy.GetCertificateSerialNumber(), proxy.String())
+				return nil, err
+			}
+
+			clusters = append(clusters, cluster)
+		}
+		return removeDups(proxy, clusters), nil
```

Comment:
Can we just use a mapset or map[string]struct{} and deduplicate by cluster.Name inline?

Reply:
this was just re-using logic that already existed. It's done in 2 separate places so figured i'd pull it out into a separate function.

Reply:
It makes sense now that I see why you did it this way (early return from function for gateway)